syntax = "proto3";

package MessageContent;

import "math.proto";

message Actor {
    string id = 1;
}

message Node {
    string id = 1;
    string name = 2;
}

message CreateActor {
    string id = 1;
    string entity = 2;
    string variant = 3;
    MathMessage.Vector position = 4;
    MathMessage.Quaternion rotation = 5;
}

message MoveActor {
    string id = 1;
    MathMessage.Vector position = 2;
    MathMessage.Quaternion rotation = 3;
}

message MoveNode {
    string id = 1;
    string name = 2;
    MathMessage.Vector position = 3;
    MathMessage.Quaternion rotation = 4;
}

message SetAnimation {
    string id = 1;
    string animation = 2;
    string file = 3;
    float speed = 4;
}

message ApplyPhysics {
    string id = 1;
    MathMessage.Vector direction = 2;
}

message LookAt {
    string id = 1;
    string target = 2;
    float distance = 3;
    MathMessage.Vector direction = 4;
}
